drivers/isl29125/isl29125.c

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright 2015 Ludwig Ortmann
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     driver_isl29125
+ * @{
+ *
+ * @file
+ * @brief       Device driver implementation for the ISL29125 RGB light sensor
+ *
+ * @author      Ludwig Ortmann <ludwig.ortmann@fu-berlin.de>
+ *
+ * @}
+ */
+
+#include <stdint.h>
+
+#include "isl29125.h"
+#include "isl29125-internal.h"
+#include "periph/i2c.h"
+#include "periph/gpio.h"
+#include "color.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+/***********************************************************************
+ * public API implementation
+ **********************************************************************/
+
+int isl29125_init(isl29125_t *dev, i2c_t i2c, uint8_t address, gpio_t gpio,
+        isl29125_mode_t mode, isl29125_range_t range,
+        isl29125_resolution_t resolution)
+{
+    int ret;
+
+    DEBUG("isl29125_init\n");
+
+    /* initialize device descriptor */
+    dev->i2c = i2c;
+    dev->address = address;
+    dev->res = resolution;
+    dev->range = range;
+    dev->gpio = gpio;
+
+    /* configuration 1: operation mode, range, resolution */
+    char conf1 = 0x00;
+    conf1 |= mode;
+    conf1 |= range;
+    conf1 |= resolution;
+    conf1 |= ISL29125_CON1_SYNCOFF; /* TODO: implement SYNC mode configuration */
+
+    /* TODO: implement configuration 2: infrared compensation configuration */
+
+    /* TODO: implement configuration 3: interrupt mode configuration */
+
+    /* acquire exclusive access to the bus */
+    DEBUG("isl29125_init: i2c_acquire\n");
+    i2c_acquire(dev->i2c);
+
+    /* initialize the I2C bus */
+    DEBUG("isl29125_init: i2c_init_master\n");
+    i2c_init_master(i2c, I2C_SPEED_NORMAL);
+
+    /* verify the device ID */
+    DEBUG("isl29125_init: i2c_read_reg\n");
+    char reg_id;
+    ret = i2c_read_reg(dev->i2c, dev->address, ISL29125_REG_ID, &reg_id);
+    if ((reg_id == ISL29125_ID) && (ret == 1)) {
+        DEBUG("isl29125_init: ID successfully verified\n");
+    }
+    else {
+        DEBUG("isl29125_init: ID could not be verified, ret: %i\n", ret);
+        ret = -1;
+        goto release;
```

Comment:
personally I do not like the use of 'goto'. But if I am the only one objecting, then leave it...